ast/src/analyzed/mod.rs

```diff
@@ -19,7 +21,7 @@ use crate::SourceRef;
 
 use self::types::TypedExpression;
 
-#[derive(Debug, Clone, Serialize, Deserialize)]
+#[derive(Debug, Clone, Serialize, Deserialize, JsonSchema)]
```

Comment:
I'm a bit wary of all these changes in the ast crate, that are specific to the versioning schema. Can't we just use the serialization that is already available for Analyzed, and only apply the schema stuff in the powdr-schema crate?

Reply:
It would be theoretically possible, but it means using [serde-reflection](https://github.com/zefchain/serde-reflection) that requires to do the reflection at runtime with a given value, and because of the nature of enums and how they are serialized, it means to keep a piece of code somewhere with every enums present in the serialized data. 
This defeat the purpose of automatically-generated schema, as one has to keep this collection of enums up to date like [here](https://github.com/diem/diem/blob/main/testsuite/generate-format/src/diem.rs#L47) for example

Reply:
Note that `schemars` also have a similar runtime-based reflection feature, but it suffers from the same drawbacks
[https://graham.cool/schemars/examples/9-from_value/](https://graham.cool/schemars/examples/9-from_value/)

Reply:
Right, ok. I think our usage of `Analyzed` as the "type of the circuit constraints" isn't great in the first place, so let's keep it like this for now and maybe one day we can refactor it into a better circuit data structure.